http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -306,48 +267,76 @@ log 'Starting up the Jupyter...'
 fi
 if [ ! -z "$RSTUDIO_DOCKER_IMAGE" ] ; then
-  COMPOSE_FILES+=(-f /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`)
-  cat /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`
+  COMPOSE_FILES+=(-f ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${RSTUDIO_DOCKER_COMPOSE}`)
+  cat ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${RSTUDIO_DOCKER_COMPOSE}`
 fi
 if [ ! -z "$WELDER_DOCKER_IMAGE" ] && [ "$WELDER_ENABLED" == "true" ] ; then
-  COMPOSE_FILES+=(-f /etc/`basename ${WELDER_DOCKER_COMPOSE}`)
-  cat /etc/`basename ${WELDER_DOCKER_COMPOSE}`
+  COMPOSE_FILES+=(-f ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${WELDER_DOCKER_COMPOSE}`)
+  cat ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${WELDER_DOCKER_COMPOSE}`
 fi
 # Note: crypto-detector should be started after user containers
 if [ ! -z "$CRYPTO_DETECTOR_DOCKER_IMAGE" ] ; then
-  COMPOSE_FILES+=(-f /etc/`basename ${CRYPTO_DETECTOR_DOCKER_COMPOSE}`)
-  cat /etc/`basename ${CRYPTO_DETECTOR_DOCKER_COMPOSE}`
+  COMPOSE_FILES+=(-f ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${CRYPTO_DETECTOR_DOCKER_COMPOSE}`)
+  cat ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${CRYPTO_DETECTOR_DOCKER_COMPOSE}`
 fi
 
-docker-compose "${COMPOSE_FILES[@]}" config
-retry 5 docker-compose "${COMPOSE_FILES[@]}" pull
-docker-compose "${COMPOSE_FILES[@]}" up -d
+if [ "${GPU_ENABLED}" == "true" ] ; then
+  COMPOSE_FILES+=(-f ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${GPU_DOCKER_COMPOSE}`)
+  cat ${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${GPU_DOCKER_COMPOSE}`
+fi
 
-# done docker-compose
-STEP_TIMINGS+=($(date +%s))
+tee /var/variables.env << END
+CERT_DIRECTORY=${CERT_DIRECTORY}
+WORK_DIRECTORY=${WORK_DIRECTORY}
+PROXY_SERVER_NAME=${PROXY_SERVER_NAME}
+PROXY_DOCKER_IMAGE=${PROXY_DOCKER_IMAGE}
+GOOGLE_PROJECT=${GOOGLE_PROJECT}
+RUNTIME_NAME=${RUNTIME_NAME}
+PROXY_SERVER_HOST_NAME=${PROXY_SERVER_HOST_NAME}
+JUPYTER_SERVER_NAME=${JUPYTER_SERVER_NAME}
+JUPYTER_DOCKER_IMAGE=${JUPYTER_DOCKER_IMAGE}
+NOTEBOOKS_DIR=${NOTEBOOKS_DIR}
+OWNER_EMAIL=${OWNER_EMAIL}
+WELDER_ENABLED=${WELDER_ENABLED}
+MEM_LIMIT=${MEM_LIMIT}
+WELDER_SERVER_NAME=${WELDER_SERVER_NAME}
+WELDER_DOCKER_IMAGE=${WELDER_DOCKER_IMAGE}
+STAGING_BUCKET=${STAGING_BUCKET}
+WELDER_MEM_LIMIT=${WELDER_MEM_LIMIT}
+CRYPTO_DETECTOR_SERVER_NAME=${CRYPTO_DETECTOR_SERVER_NAME}
+CRYPTO_DETECTOR_DOCKER_IMAGE=${CRYPTO_DETECTOR_DOCKER_IMAGE}
+JUPYTER_SCRIPTS=${JUPYTER_SCRIPTS}
+HOST_PROXY_SITE_CONF_FILE_PATH=${DOCKER_COMPOSE_FILES_DIRECTORY}/`basename ${PROXY_SITE_CONF}`
+DOCKER_COMPOSE_FILES_DIRECTORY=${DOCKER_COMPOSE_FILES_DIRECTORY}
+END
 
-# If Welder is installed, start the service.
-# See https://broadworkbench.atlassian.net/browse/IA-1026
-if [ ! -z "$WELDER_DOCKER_IMAGE" ] && [ "$WELDER_ENABLED" == "true" ] ; then
-  log 'Starting Welder (file synchronization service)...'
-  retry 3 docker exec -d ${WELDER_SERVER_NAME} /opt/docker/bin/entrypoint.sh
-fi
+# Create a network that allows containers to talk to each other via exposed ports
+docker network create -d bridge app_network
```

Comment:
Wonder if it's possible to create the network in `network-compose.yaml`? (thinking out loud, not a big deal either way)

Reply:
the network needs to exist before other services can reference it...but `network.yaml` needs to be appended last because it references other services..so I think we'll have to create it beforehand.